scripts/upgrade-facets.js

```diff
@@ -0,0 +1,244 @@
+const hre = require("hardhat");
+const ethers = hre.ethers;
+const network = hre.network.name;
+const { Facets } = require("./config/facet-upgrade");
+const { readContracts } = require("./util/utils");
+const environments = require("../environments");
+const confirmations = network == "hardhat" ? 1 : environments.confirmations;
+const tipMultiplier = ethers.BigNumber.from(environments.tipMultiplier);
+const tipSuggestion = "1500000000"; // ethers.js always returns this constant, it does not vary per block
+const maxPriorityFeePerGas = ethers.BigNumber.from(tipSuggestion).mul(tipMultiplier);
+const { deployProtocolHandlerFacets } = require("./util/deploy-protocol-handler-facets.js");
+const { FacetCutAction, getSelectors, removeSelectors } = require("./util/diamond-utils.js");
+const { deploymentComplete, getFees, writeContracts } = require("./util/utils.js");
+const Role = require("./domain/Role");
+const packageFile = require("../package.json");
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+/**
+ * Upgrades or reamoves existing facets, or add new facets.
+ *
+ * Process:
+ *  1.  Edit scripts/config/facet-upgrade.js.
+ *  1a. Provide a list of facets that needs to be upgraded (field "addOrUpgrade") or removed completely (field "remove")
+ *  1b. Optionally you can specify which selectors should be ignored (filed "skip"). You don't have to specify "initialize()" since it's ignored by default
+ *  2. Update protocol version in package.json. If not, script will prompt you to confirm that version remains unchanged.
+ *  2. Run the appropriate npm script in package.json to upgrde facets for a given network
+ *  3. Save changes to the repo as a record of what was upgraded
+ */
+async function main() {
+  // Bail now if hardhat network
+  if (network === "hardhat") process.exit();
+
+  const chainId = (await hre.ethers.provider.getNetwork()).chainId;
+  const contractsFile = readContracts(chainId, network);
+  let contracts = contractsFile.contracts;
+
+  const divider = "-".repeat(80);
+  console.log(`${divider}\nBoson Protocol Contract Suite Upgrader\n${divider}`);
+  console.log(`⛓  Network: ${hre.network.name}\n📅 ${new Date()}`);
+
+  // Check that package.json version was updated
+  if (packageFile.version == contractsFile.protocolVersion) {
+    const answer = await new Promise((resolve) => {
+      rl.question("Protocol version has not been updated. Proceed anyway? (y/n) ", resolve);
+    });
+    switch (answer.toLowerCase()) {
+      case "y":
+      case "yes":
+        break;
+      case "n":
+      case "no":
+        process.exit(1);
+        break;
+      default:
+        break;
```

Comment:
I would make the default do `process.exit()` as well. If they didn't give a yes or no answer, then we should just quit.